python/cuml/common/handle.pyx

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+
+from libcpp.memory cimport shared_ptr
+from cuml.common.cuda cimport _Stream
+
+
+cdef extern from "common/rmmAllocatorAdapter.hpp" namespace "ML" nogil:
+    cdef cppclass rmmAllocatorAdapter(deviceAllocator):
+        pass
+
+
+# TODO: name this properly!
```

Comment:
Wouldn't it be better to have the discussion for the right name now? Other code will pick this up so the later we do this the more work it will be to change it.

Reply:
Yes, please. Suggestions for a better name will be appreciated :)

Reply:
How would a usage of the current proposal look in python? I.e. is it part as something like a namespace?

Reply:
1. the simple unit-test `python/cuml/test/test_base.py` has one way.
2. Have updated documentation of `base.pyx`, with an example usage, too.

Reply:
If you would omit `from cuml import Base, Handle` in `python/cuml/test/test_base.py` it would be

```python
handle = cuml.Handle()
```

instead of 

```python
handle = Handle()
```

right? In this case what is wrong with the name `Handle`? It is unambiguous via the module name. I must admit that I do not have a better suggestion.

Reply:
That's correct, it'll become `cuml.Handle`.

I too couldn't come up with a better name. Hence, dropped this todo note for reviewers. Seems like we can keep this name after all! So, now it makes sense to remove this comment :)

Reply:
you made a good point, BTW. It's cleaner as an example to omit the `from cuml ...` statement from the unit-test file. Let me do that.

Reply:
Done. Made those changes. I'm resolving this conversation @jirikraus , for now.